capstone/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -51,8 +51,8 @@
   private final String CREATE_NEW_USER_QUERY =
     "INSERT INTO user_info (user_name, full_name, email, verified, university_id) VALUES (?, ?, ?, true, ?)";
 
-  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_OUT = "";
-  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_IN = "";
+  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_OUT = "/";
+  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_IN = "/users?query_source=server";
```

Comment:
I'm having trouble understanding what the `?query_source=server` is trying to do. Explain the problem you're trying to solve and I suspect we can come up with a simpler way to solve it.

Reply:
I noticed, when I called the "/users" endpoint from javascript using the fetch command, it neglects the sendRedirect() line in my UserServlet.java file, so I used this "?query_source=server" and also in my fetch I added "?query_source=client" to differentiate if the call to the user endpoint is from the JS fetch command or not. Based on that, I'll either run the sendRedirect() line or return the userDetails JSON value.

Reply:
Gotcha. The way a redirect typically works is the server responds with a 301 or 307 status code and sets the "Location" header to the new URL. You can see this with `curl -v google.com`:

```
...
< Location: http://www.google.com/
...
```

Could you have the handler return a 307 and set the Location header to the login page if user is not logged in, then the JS can fire the redirect to the value of Location header on a 307. Else on a 200 the JS reads the response body to get the user info.

As a side note, I'd probably rename the endpoint to `GET /users/current` as `GET /users` looks like it would return a list of all users.

Reply:
I understand what you explained, but I'm confused on how to implement it to what achieve the same result I currently have.